src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -440,26 +430,23 @@ private static LogEntry WriteEntry(EventLevel level, Exception innerException, s
                     message = string.Format(CultureInfo.InvariantCulture, "Message: {0}, InnerException: {1}. ", message, innerException.Message);
             }
 
+            message = args == null ? message : FormatInvariant(message, args);
+
             // Logs basic information (library version, DateTime, whether PII is ON/OFF) once before any log messages are written.
             if (!_isHeaderWritten)
             {
                 _isHeaderWritten = true;
```

Comment:
>_isHeaderWritten = true; [](http://example.com/codeflow?start=16&length=24)

should this get set after the Logger.Log call is made? Seems like better to err or the side of double logging this as opposed to missing it right? I might be missing what prevents it but seems like even this way there's a potential for double logging in race conditions.